SharpBucket/V2/EndPoints/DiffParameters.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SharpBucket.V2.EndPoints
+{
+    /// <summary>
+    /// Parameters for RepositoryResource.GetDiff().
+    /// </summary>
+    public class DiffParameters: ParametersBase
+    {
+        /// <summary>
+        /// Number of context lines for each difference; default is 3.
+        /// </summary>
+        public byte Context { get; set; } = 3;
+        /// <summary>
+        /// The paths of specific files to diff.
+        /// </summary>
+        public IEnumerable<string> Paths { get; set; } = new List<string>();
+        /// <summary>
+        /// Generate diffs that ignore whitespace; default is false.
+        /// </summary>
+        public bool IgnoreWhitespace { get; set; }
+        /// <summary>
+        /// Generate diffs that include binary files; default is true.
+        /// </summary>
+        public bool Binary { get; set; } = true;
+
+        internal override IDictionary<string, object> ToDictionary()
+        {
+            return DictionaryFromKvps(
```

Comment:
I'm wondering if it would not have been better to improve the `ObjectToDictionaryHelper` class by allowing to ignore properties which values are equivalent to their default values (which can be defined with the `DefaultValueAttribute`) instead of writing custom `ToDictionary` code here and in `ListParameters`

like that there is no need to explicitly convert our request parameters objects to dictionaries before given them to the `internal T Get<T>(T body, string overrideUrl, object requestParameters = null)` method of `SharpBucket` class

Reply:
There are a couple of tricky things here:

1. The query string parameter name is not always the same as the name of the property on the Parameters object. For example, on `ListParameters` the `Filter` property needs to be `q=` in the query string. Yes, this could also be accomplished with custom attributes and reflection, but this approach seemed cleaner.
2. For the boolean properties, BB wants the literal values `true` and `false`, but the string value of the boolean `true` is `True`. Handling this within the parameters object lets me specify exactly what I want.

Reply:
Your second tricky point is interesting because that means that each time we will want to write a class like that one we will have to know that Bitbucket expect boolean to be serialized in lowercase.

So is it more conformatable to have to manage that explicitly each time we will do that ?
Or should we more go like I suggested in an improvement of the `ObjectToDictionaryHelper` to write once for all the code needed to handle that conventions of conversions ?

Reply:
@MitjaBezensek can you give us your opinion on that part of the code ?

Reply:
Well, there could also be a case when BB wants true/false to be represented as 1/0. 

So I think there are a couple of options. My vote would be to leave things as I have them. Each `Parameters` class is responsible for turning itself into a dictionary.  Unless we start having other objects that have similar specialized needs for turning themselves into dictionaries, I think this is the best solution and gives the most control over the process.

The other option would be to add a method to the `ObjectToDictionaryHelper`. We would need at least two new custom atttibutes for the `Parameters` classes (for the key name and the format of the value), and then the new method in the helper would need to know how to deal with those attributes (as well as `[DefaultValue]`). To me, this seems like making the logic too global, when it's only needed by these `Parameters` objects. They should know how to turn themselves into dictionaries.

Reply:
I'm all for the logic handling these cases on its own, without the contributors having to need to know these details (like @mnivet suggests), but it seems that in this case, it would bring quite some additional complexity. 

I would say let's go with @asherber's suggestion. For now, it seems to offer the most flexibility, on top of that the logic is mostly internal logic and would not affect the end users of the library even if found that we would need to change it (don't think people will start using `ToDictionary` method, but you never know).

Reply:
OK let's go with current solution and merge that pull request then